enhancements/network/network-policy-audit-logging.md

```diff
@@ -0,0 +1,357 @@
+ // +optional
+ MTU *uint32 `json:"mtu,omitempty"`
+ // geneve port is the UDP port to be used by geneve encapulation.
+ // Default is 6081
+ // +kubebuilder:validation:Minimum=1
+ // +optional
+ GenevePort *uint32 `json:"genevePort,omitempty"`
+ // HybridOverlayConfig configures an additional overlay network for peers that are
+ // not using OVN.
+ // +optional
+ HybridOverlayConfig *HybridOverlayConfig `json:"hybridOverlayConfig,omitempty"`
+ // ipsecConfig enables and configures IPsec for pods on the pod network within the
+ // cluster.
+ // +optional
+ IPsecConfig *IPsecConfig `json:"ipsecConfig,omitempty"`
+ 
+ <--- BEGIN NEW CODE --->
+ // aclLoggingRateLimit allows for the configuring of max acl logging rate
+ // defaults to 20 messages per second
+ // +optional
+ aclLoggingRateLimit  *uint32 `json:"aclLoggingRateLimit,omitempty"`
+ <--- END NEW CODE --->
+
+}
+```
+
+The flag `aclLoggingRateLimit` defaults to `20` messages per second.
+
+This will result in an updated OVNKubernetes Operator configuration object: `(.spec.defaultNetwork.ovnKubernetesConfig)` which will enable cluster-wide configuration of ACL logging at cluster installation time:
+
+```go
+spec:
+  defaultNetwork:
+    type: OVNKubernetes
+    ovnKubernetesConfig:
+      mtu: 1400
+      genevePort: 6081
+      ipsecConfig: {}
+      aclLoggingRateLimit: 20
+```
+
+To enable the ACL logging on a per namespace basis using metadata annotations in the namespace's definition like the following
+
+```go
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: tenantA
+  annotations:
+    k8s.ovn.org/acl-logging: '{ "deny": "alert", "allow": "notice" }'
+```
+
+or
+
+```go
+kind: Namespace
+apiVersion: v1
+metadata:
+  name: tenantB
+  annotations:
+    k8s.ovn.org/acl-logging: '{ "deny": "notice" }'
+```
+
+The logging can be activated for either allow, drop, or allow and drop actions. The severity must be one of `alert`, `warning`,
+`notice`, `info`, or `debug` as described in [OVN documentation](http://www.openvswitch.org/support/dist-docs/ovn-nbctl.8.html).
+
+each level is listed in order or descending severity
+
+```go
+alert    A major failure forced a process to abort.
+
+warning  A high-level operation or a subsystem failed.  Attention
+         is warranted.
+
+notice   A low-level operation failed, but higher-level subsystems
+         may be able to recover.
+
+info     Information that may be useful in retrospect when
+         investigating a problem.
+
+debug    Information useful only to someone with intricate
+         knowledge of the system, or that would commonly cause too-
+         voluminous log output.  Log messages at this level are not
+         logged by default.
+```
+
+### User Stories
+
+#### Story 1
+
+As an Openshift user I want to see what Network Policies and accompanied ACLs are dropping the most traffic in a cluster. For example, when a packet runs into a `drop` ACL it will log a message similar to the following:
+
+ ```go
```

Comment:
nit: not actually golang ... maybe sh?  I'm not sure what is best for quoting a log message ...

@dhellmann any ideas for what's appropriate here since you did the cleanup for turning on linting?